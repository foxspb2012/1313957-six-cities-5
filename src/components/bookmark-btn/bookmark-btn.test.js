import React from 'react';
import renderer from 'react-test-renderer';
import BookmarkButton from './bookmark-btn';

const TYPE = `place-card`;

const testing = () => {};

describe(`Should BookmarkButton render correctly`, () => {
  it(`Should BookmarkButton render correctly with isActive`, () => {
    const tree = renderer
      .create(
          <BookmarkButton
            isActive={true}
            typeClass={TYPE}
            onClick={testing}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should BookmarkButton render correctly without isActive`, () => {
    const tree = renderer
      .create(
          <BookmarkButton
            isActive={true}
            typeClass={TYPE}
            onClick={testing}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
