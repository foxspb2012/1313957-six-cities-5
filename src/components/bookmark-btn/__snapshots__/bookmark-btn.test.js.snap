// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should BookmarkButton render correctly Should BookmarkButton render correctly with isActive 1`] = `
<button
  className="undefined__bookmark-button button undefined__bookmark-button--active"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;

exports[`Should BookmarkButton render correctly Should BookmarkButton render correctly without isActive 1`] = `
<button
  className="undefined__bookmark-button button undefined__bookmark-button--active"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;
